src/io.jl

```diff
@@ -95,16 +95,20 @@ Saves the solution variable v_investment to a file "investments.csv" inside `out
 The format of each row is `a,v,p*v`, where `a` is the asset indexing `v_investment`, `v`
 is corresponding `v_investment` value, and `p` is the corresponding `p_unit_capacity` value.
 """
-function save_solution_to_file(output_folder, v_investment, p_unit_capacity)
+function save_solution_to_file(
+    output_folder,
+    s_assets_investment,
+    v_investment,
+    p_unit_capacity,
+)
     # Writing the investment results to a CSV file
-    output_file = open(joinpath(output_folder, "investments.csv"), "w")
-    write(output_file, "a,InstalUnits,InstalCap_MW\n")
-    for a in axes(v_investment, 1)
-        v = v_investment[a]
-        p = p_unit_capacity[a]
-        write(output_file, "$a,$v,$(p * v)\n")
-    end
-    close(output_file)
+    output_file = joinpath(output_folder, "investments.csv")
+    output_table = DataFrame(;
+        a = s_assets_investment,
+        InstalUnits = [v_investment[a] for a in s_assets_investment],
+        InstalCap_MW = [p_unit_capacity[a] * v_investment[a] for a in s_assets_investment],
```

Comment:
Is the comma at the end of this line correct?

Reply:
Yes, although it is optional and part of the formatting style: https://domluna.github.io/JuliaFormatter.jl/stable/#trailing_comma
The language understands that this last comma is part of style, and doesn't complain about it.

Trailing commas in multiline expressions (arrays, function arguments) are used with a bit of git in mind.
If you add an additional argument to that list, git will say you made two changes: 1. Add a comma at line N. 2. Add the new line.

Example: Let's say you have an array `v = [1, 2]` and changes it to `v = [1, 2, 3]`. Also assume that the actual array has long enough arguments that warrant breaking it into multiple lines.
If you don't have a comma:

```diff
 v = [
   1,
-  2
+  2,
+  3
]
```

If you have a comma:

```diff
 v = [
   1,
   2,
+  3,
]
```

The other argument for it is that if you reorder things, you don't have to fix that comma.

It is, of course, unnecessary and opinionated.

Reply:
This is super handy!